commit f2c89c67d8d7060ef6aa6491815fb7dec3426fc2
Author: Gottfried Haider <gottfried.haider@gmail.com>
Date:   Tue Sep 26 11:59:51 2017 +0000

    pwm: Set class for exported channels in sysfs
    
    [ Upstream commit 7e5d1fd75c3dde9fc10c4472b9368089d1b81d00 ]
    
    Notifications for devices without bus or class set get dropped by
    dev_uevent_filter(). Adding the class to the exported child matches
    what the GPIO subsystem is doing.
    
    With this change exporting a channel triggers a udev event, which
    gives userspace a chance to fixup permissions and makes it possible
    for non-root users to make use of the PWM subsystem.
    
    Signed-off-by: Gottfried Haider <gottfried.haider@gmail.com>
    CC: Thierry Reding <thierry.reding@gmail.com>
    CC: H Hartley Sweeten <hsweeten@visionengravers.com>
    CC: linux-pwm@vger.kernel.org
    CC: linux-arm-kernel@lists.infradead.org
    CC: linux-rpi-kernel@lists.infradead.org
    Signed-off-by: Thierry Reding <thierry.reding@gmail.com>

diff --git a/drivers/pwm/sysfs.c b/drivers/pwm/sysfs.c
index a813239300c3..83f2b0b15712 100644
--- a/drivers/pwm/sysfs.c
+++ b/drivers/pwm/sysfs.c
@@ -263,6 +263,7 @@ static int pwm_export_child(struct device *parent, struct pwm_device *pwm)
 	export->pwm = pwm;
 	mutex_init(&export->lock);
 
+	export->child.class = parent->class;
 	export->child.release = pwm_export_release;
 	export->child.parent = parent;
 	export->child.devt = MKDEV(0, 0);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1153_f2c89c67d8d7060ef6aa6491815fb7dec3426fc2.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1153_f2c89c67d8d7060ef6aa6491815fb7dec3426fc2.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
